src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -650,15 +685,9 @@ def command(self, name, method_name=None, command_type=None, **kwargs):
         :rtype: None
         """
         self._check_stale()
-        merged_kwargs = self.group_kwargs.copy()
-        group_command_type = merged_kwargs.get('command_type', None)
-        if command_type:
-            merged_kwargs.update(command_type.settings)
-        elif group_command_type:
-            merged_kwargs.update(group_command_type.settings)
-        merged_kwargs.update(kwargs)
 
-        operations_tmpl = merged_kwargs.get('operations_tmpl')
+        merged_kwargs = self._merge_kwargs(kwargs)
+        operations_tmpl = self._resolve_kwarg(merged_kwargs, 'operations_tmpl', required=True)
```

Comment:
We should do the final merging of the command_type kwargs and previous merged_kwargs here, where the merged_kwargs take precedence..
This way we don't need to call _resolve_kwarg() or deal with juggling two sets of kwargs after this point (like we do with 'client_factory' in `_cli_command()`).

Reply:
Yes, now I just have a flatten operation that merges everything and do away with resolve_kwargs.